### PR TITLE
Give event timestamps microsecond precision (#1694)

### PR DESCRIPTION
**Pull request by njhale**
The `metav1.Time` type used as timestamps in acorn events were
being truncated to seconds precision when serialized. This
limitation meant that we didn't have the ability to order events
that occur within the same second; e.g. resource contention between
controllers, flapping, etc.

To this end, switch to `metadata.MicroTime`.

Signed-off-by: Nick Hale <4175918+njhale@users.noreply.github.com>

### CODE REVIEW

File: pkg/apis/internal.acorn.io/v1/event.go [hunk 1]
```diff
@@ -40,7 +40,7 @@ type EventInstance struct {
 	Description string `json:"description,omitempty"`
 
 	// Observed represents the time the Event was first observed.
-	Observed metav1.Time `json:"observed"`
+	Observed metav1.MicroTime `json:"observed" wrangler:"type=string"`
```
Comment (g-linville) [inline pkg/apis/internal.acorn.io/v1/event.go:43]: Just curious, why did you need to add this `wrangler` tag to this struct field?

**Review by g-linville — APPROVED** [review]

File: pkg/apis/internal.acorn.io/v1/event.go (hunk 1 above)
Comment (njhale) [reply pkg/apis/internal.acorn.io/v1/event.go:43]: We use `wrangler` to generate the internal CRDs at runtime, and without this, it generates an incorrect openapi spec for `MicroTime` that breaks serialization and conversion; general chaos.

**Review by thedadams — APPROVED** [review]
